components/html-block/html-block.js

```diff
@@ -250,8 +267,21 @@ class HtmlBlock extends RtlMixin(LitElement) {
 	}
 
 	async _handleSlotChange(e) {
-		if (!e.target) return;
-		await this._render(e.target);
+		if (!e.target || !this.shadowRoot) return;
+		const slot = this.shadowRoot.querySelector('slot');
+		const slottedNodes = slot.assignedNodes({ flatten: true });
+
+		if (!this.html && this._hasSlottedElements(slottedNodes)) {
```

Comment:
The addition of the `!this.html` check here will need some rethinking when consumers have implemented the new API. For now, however, it's required to maintain consistent behaviour. While the html-block doesn't currently _fully_ work properly if it contains only non-element nodes, the initial render does still work. Without this additional check, it stops working, and that's not good. While "partially broken" still isn't ideal, "completely broken" is worse.

Reply:
Scratch this. It wasn't sufficient and I found a better way to handle this.

Reply:
I'm dumb. It wasn't sufficient, but it _was_ still necessary. 🤦